core/src/dird/ua_status.cc

```diff
@@ -513,46 +532,58 @@ static bool DoSubscriptionStatus(UaContext* ua)
     return false;
   }
 
+  char now[30] = {0};
+  bstrftime(now, sizeof(now), (utime_t)time(NULL), "%F %T");
+
   if (kw_all || kw_detail) {
+    ua->send->ObjectKeyValue(
+        "version", "Bareos version: ", kBareosVersionStrings.Full, "%s");
+    ua->send->ObjectKeyValue("os", kBareosVersionStrings.GetOsInfo(),
+                             " (%s)\n");
+    ua->send->ObjectKeyValue("binary-info",
+                             "Binary info: ", kBareosVersionStrings.BinaryInfo,
+                             "%s\n");
+    ua->send->ObjectKeyValue("report-time", "Report time: ", now, "%s\n");
     ua->SendMsg(_("\nDetailed backup unit report:\n"));
-    ua->send->ObjectStart("unit-detail");
     ua->db->ListSqlQuery(
         ua->jcr,
         BareosDb::SQL_QUERY::subscription_select_backup_unit_overview_0,
-        ua->send, HORZ_LIST, true);
-    ua->send->ObjectEnd("unit-detail");
+        ua->send, HORZ_LIST, "unit-detail", true);
   }
   if (kw_all || kw_detail || !kw_unknown) {
     ua->SendMsg(_("\nBackup unit totals:\n"));
-    ua->send->ObjectStart("total-units-required");
-
     PoolMem query(PM_MESSAGE);
     ua->db->FillQuery(
         query, BareosDb::SQL_QUERY::subscription_select_backup_unit_total_1,
         me->subscriptions);
-    ua->db->ListSqlQuery(ua->jcr, query.c_str(), ua->send, VERT_LIST, true);
-    ua->send->ObjectEnd("total-units-required");
+    ua->db->ListSqlQuery(ua->jcr, query.c_str(), ua->send, VERT_LIST,
+                         "total-units-required", true,
+                         BareosDb::CollapseMode::Collapse);
+    std::string checksum_source
+        = get_subscription_status_checksum_source_text(ua, now);
+    std::optional<std::string> checksum = compute_hash(checksum_source.c_str());
+    if (checksum) {
+      ua->send->ObjectKeyValue("checksum",
+                               "Checksum: ", checksum.value().c_str(), "%s\n");
+    }
```

Comment:
* i would probably have used `auto` instead of `std::optional<std::string>` here.
* `compute_hash()` takes a `const std::string&` as first parameter. You're passing the underlying `const char*` of your existing `std::string` to it, which will lead to an implicit conversion to std::string() which implies another copy. Just passing the `std::string` would be better
* there are two ways to get the value out of a `std::optional`: calling `value()` or `operator*`/`operator->`. With `value()` you get a check that there is a value in your optional (and an exception if there is none), while with the operators behaviour is undefined if there is no value. As you checked that there is a value, you can safely use the operators, simplifying `checksum.value().c_str()` to just `checksum->c_str()`.